### ISSUE TITLE
misuse of `strlit` not caught

### ISSUE THREAD

**Issue by nano-o** [microsoft/ivy#74]
Hello, the generated code for `impl` crashes at runtime with the following error, although `ivy_check` does not complain.
```
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_M_construct null not valid
```
```
#lang ivy1.7

type t
interpret t -> strlit
individual x:t
action my_action(x:t) = {
}
after init {
    call my_action(0);
}

extract impl = this
```

**Comment by kenmcmil** [comment]
Fixed in commit 4d97c7e. Now outputs this:

````
error: Cannot compile numeral 0:t of string sort t
````

Thanks.

**Comment by kenmcmil** [comment]
Oops. Pushed fix for the fix.
